docs/notebooks/burger.ipynb

```diff
@@ -0,0 +1,745 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Adjoint-based gradient\n",
+    "We seek to compute the sensitivity/gradient $\\mathrm{d}\\widehat{I}/\\mathrm{d}u_0$, where $\\widehat{I} = I(u_0)$ is the reduced functional. On applying the chain rule, we have the below expression on considering the functional $I$ as in Eq. (1).\n",
+    "$$\\frac{\\mathrm{d}\\widehat{I}}{\\mathrm{d}u_0} = \\frac{\\partial I}{\\partial u} \\frac{\\mathrm{d}u}{\\mathrm{d}u_0}. \\tag{3}$$\n",
+    "Computing $\\partial J/\\partial u$ is straightforward since the functional (Eq. (1)) is written in terms of the solution variable $u$. In contrast, the Jacobian $\\mathrm{d}u/\\mathrm{d}u_0$ involves an expensive computational operation that is avoided by using the adjoint method.\n",
+    "\n",
+    "Taking the adjoint (Hermitian transpose) of the above equation:\n",
+    "$$\\frac{\\mathrm{d}\\widehat{I}}{\\mathrm{d}u_0}^* = - \\frac{\\partial F}{\\partial u_0}^* \\frac{\\partial F}{\\partial u}^{-*} \\frac{\\partial I}{\\partial u}^{*}$$\n",
+    "\n",
+    "Next, we define:\n",
+    "$$\\lambda = \\left(\\frac{\\partial F(u, u_0)}{\\partial u}\\right)^{-*} \\frac{\\partial I}{\\partial u}^* \n",
+    "  \\implies  \\left(\\frac{\\partial F(u, u_0)}{\\partial u}\\right)^{*} \\lambda = \\frac{\\partial I}{\\partial u}^*,$$\n",
+    "which is the adjoint system. Finally, we obtain the following expression for the adjoint-based gradient computation:\n",
+    "$$\\frac{\\mathrm{d}\\widehat{I}}{\\mathrm{d} u_0}^* = - \\frac{\\partial F}{\\partial u_0}^* \\lambda.$$\n",
+    "In this current case, Burger's equation is written in weak form and discretized in time with the backward finite difference method, we have:\n",
+    "$$ F(u^{n+1}, u^n, v) = \\int_{\\Omega} u^{n + 1} \\cdot v - u^{n}  \\cdot v + \\Delta t \\left(u^{n + 1} \\frac{\\partial u^{n + 1}}{\\partial x} \\cdot v + \\frac{\\partial u^{n + 1}}{\\partial x}  \\cdot \\frac{\\partial v}{\\partial x}\\right) \\, dx = 0 \\quad \\forall v\\in V \\tag{4},$$\n",
+    "where $v \\in V$ is an arbitrary test function. \n",
+    "\n",
+    "The time sequence of the adjoint system for the functional (Eq. (1)) is given by:\n",
+    "$$\n",
+    "\\begin{align*}\n",
+    "    \\lambda^{N+1} &= \\frac{\\partial I}{\\partial u^{N + 1}}^{\\ast}\\\\\n",
+    "    \\\\\n",
+    "    \\frac{\\partial F}{\\partial u^{N + 1}}^* \\lambda^{N} &= \\lambda^{N+1}\n",
+    "    \\\\\n",
+    "    \\frac{\\partial F}{\\partial u^{N}}^* \\lambda^{N - 1} &= \\lambda^{N} \n",
+    "    \\\\\n",
+    "    \\vdots \\\\\n",
+    "    \\frac{\\partial F}{\\partial u^{1}}^* \\lambda^{0} &= \\lambda^{1}\n",
+    "\\end{align*}\n",
+    "\\tag{5}\n",
+    "$$\n",
+    "where $\\lambda^{n}$ is the adjoint variable at time step $n$.\n",
+    "\n",
+    "For additional details on the discrete adjoint-based gradient computation, you may refer to the following references [1, 2]."
```

Comment:
[1] not needed if Gunzburger reference is added above.

Reply:
I removed this text and added Gunzburger reference above.